tests/testthat/test-predictions.R

```diff
@@ -9,3 +9,42 @@ test_that("always uses na.action = na.exclude", {
 
   expect_equal(2 * 2, 4)
 })
+
+test_that("prediction functions return expected outputs", {
+  df <- tibble::data_frame(
```

Comment:
I think you could simplify this quite a bit with a simpler data set, a single model, and by making assertions about the "shape" of the output, not the specific values:

```R
df <- data.frame(x = 1:5, y = c(1, 4, 3, 2, 5))
mod <- lm(y ~ x, data = df)

out <- spread_predictions(df, m1 = mod, m2 = mod)
expect_s3_class(out, "tbl_df")
expect_named(out, c("x", "y", "m1", "m2"))
expect_equal(nrow(out), nrow(df))

out <- gather_predictions(df, m1 = mod, m2 = mod)
expect_s3_class(out, "tbl_df")
expect_named(out, c("model", "x", "y", "pred"))
expect_equal(nrow(out), nrow(df) * 2)
```